src/net.rs

```diff
@@ -180,12 +172,10 @@ impl HttpListener {
     /// Start listening to an address over HTTPS.
     pub fn https<To: ToSocketAddrs>(addr: To, cert: &Path, key: &Path) -> io::Result<HttpListener> {
         let mut ssl_context = try!(SslContext::new(Sslv23).map_err(lift_ssl_error));
-        try_some!(ssl_context.set_cipher_list("DEFAULT").map(lift_ssl_error));
-        try_some!(ssl_context.set_certificate_file(
-                cert, X509FileType::PEM).map(lift_ssl_error));
-        try_some!(ssl_context.set_private_key_file(
-                key, X509FileType::PEM).map(lift_ssl_error));
-        ssl_context.set_verify(SslVerifyNone, None);
+        try!(ssl_context.set_cipher_list("DEFAULT").map_err(lift_ssl_error));
+        try!(ssl_context.set_certificate_file(cert, X509FileType::PEM).map_err(lift_ssl_error));
+        try!(ssl_context.set_private_key_file(key, X509FileType::PEM).map_err(lift_ssl_error));
```

Comment:
If that's what we want, it can be in a separate PR. Although, with the `io::Error` changes to holding a `Box<Error>`, the original SSL error is no longer really lost. It's boxed into a trait object.